galley/pkg/config/processing/snapshotter/distributor.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package snapshotter
+
+import (
+	"sync"
+
+	"istio.io/istio/pkg/mcp/snapshot"
+)
+
+// Distributor interface abstracts the snapshotImpl distribution mechanism. Typically, this is implemented by the MCP layer.
+type Distributor interface {
+	SetSnapshot(name string, s snapshot.Snapshot)
+}
+
+// InMemoryDistributor is an in-memory Distributor implementation.
+type InMemoryDistributor struct {
+	snapshotsLock sync.Mutex
+	snapshots     map[string]snapshot.Snapshot
+}
+
+var _ Distributor = &InMemoryDistributor{}
+
+// NewInMemoryDistributor returns a new instance of InMemoryDistributor
+func NewInMemoryDistributor() *InMemoryDistributor {
+	return &InMemoryDistributor{
+		snapshots: make(map[string]snapshot.Snapshot),
+	}
+}
+
+// SetSnapshot is an implementation of Distributor.SetSnapshot
+func (d *InMemoryDistributor) SetSnapshot(name string, s snapshot.Snapshot) {
+	d.snapshotsLock.Lock()
+	defer d.snapshotsLock.Unlock()
+
+	scope.Infof("InmemoryDistributor.SetSnapshot: %s: %v", name, s)
```

Comment:
info or debug?

Reply:
BTW, the reason I'm being picky about log levels is that I believe the default log data produced by a component should be geared towards the operator and not the implementor. The operator doesn't understand the implementation details of the code, such  as function names or type names. Thus, info-level log messages need to deal in operator-visible concepts and things the operator can do something about and/or cares about.

Anything more should be debug-level.

Reply:
While, I don't disagree with the sentiment, my experience has been that most people don't actually look at logs and try to understand what is going on. The logs have been useful with helping us (developers) debug issues happening on other operator's systems. 

In this case debug seems more appropriate though.